src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -154,6 +154,16 @@ Map<String, Object> buildMapWithProcessorKeyAndOriginalValue(IngestDocument inge
         for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
             String originalKey = fieldMapEntry.getKey();
             Object targetKey = fieldMapEntry.getValue();
+
+            int nestedDotIndex = originalKey.indexOf('.');
```

Comment:
This is a user-provided info, can we add basic validation if it's not already done as part of the processor/pipeline definition.

if multiple levels of nested fields are needed this code may need a rework

Reply:
There is some basic validation done in `validateEmbeddingConfiguration` which is run on `fieldMap` in the constructor in this file. However, I will add some extra validation for nested fields.